src/query.ts

```diff
@@ -216,16 +216,22 @@ export class Query {
    *                                         (default) or an "exclude" filter.
    *
    */
-  public toggleFilter(filterName: string, value: FacetOption, filterType = 'filter'): void {
-    if (!isPlainObject(value)) {
-      (value as Array<unknown>).forEach((value: any) => {
-        if (!this.hasFilter(filterName, value, filterType)) {
-          this.addFilter(filterName, [value], filterType);
-        } else {
-          this.removeFilter(filterName, [value], filterType);
-        }
-      });
+  public toggleFilter(filterName: string, value: FacetOption | string | number, filterType = 'filter'): void {
+    let values: FacetOption = [];
+
+    if (isArray(value)) {
+      values = value as FacetOption;
+    } else if (typeof value === 'string' || typeof value === 'number') {
+      (values as Array<unknown>).push(value);
     }
+
+    (values as Array<unknown>).forEach((value: any) => {
+      if (!this.hasFilter(filterName, value, filterType)) {
+        this.addFilter(filterName, [value], filterType);
+      } else {
+        this.removeFilter(filterName, [value], filterType);
+      }
+    });
```

Comment:
You know I usually prefer ask without negations, and given we have an else… This may help with types.

```typescript
values.forEach((value: RangeFacet|string|number) => {
  if (this.hasFilter(filterName, value, filterType)) {
    this.removeFilter(filterName, [value], filterType);
  } else {
    this.addFilter(filterName, [value], filterType);
  }
});
```

Not tested, but I think is more or less correct :smile: